src/main/java/tech/ydb/io/r2dbc/result/YdbResult.java

```diff
@@ -47,14 +49,39 @@ private YdbResult(Flux<RowSegment> segments, long rowsUpdated) {
     }
 
     public YdbResult(ResultSetReader resultSetReader) {
-        List<RowSegment> rowSegments = new ArrayList<>(resultSetReader.getRowCount());
+        this.rowsUpdated = DEFAULT_SELECT_ROWS_UPDATED;
+        this.segments = Flux.generate(
+                YdbRowMetadataState::new,
+                (state, sink) -> {
+                    if (!resultSetReader.next()) {
+                        sink.complete();
+                        return state;
+                    }
+                    YdbRowMetadataState currentState = state;
+                    if (!state.isInitialized()) {
+                        currentState = new YdbRowMetadataState(getYdbRowMetadata(resultSetReader));
+                    }
+                    List<Value<?>> values = new ArrayList<>(resultSetReader.getColumnCount());
+                    for (int index = 0; index < resultSetReader.getColumnCount(); index++) {
+                        values.add(resultSetReader.getColumn(index).getValue());
+                    }
+
+                    sink.next(new RowSegment(new YdbRow(currentState.getYdbRowMetadata(), values)));
+                    return currentState;
+                });
+    }
 
```

Comment:
Because YdbRow now stores a list of Value<?> in order not to block the stream from reading the entire result, I replaced the creation with Flux.generate(), there is a test to check this.

Из-за того, что теперь в YdbRow хранится список Value<?>, чтобы не блокировать поток на чтение всего результата, заменил создание на Flux.generate(), на проверку этого есть тест.